funfact/util/typing.py

```diff
@@ -2,5 +2,5 @@
 # -*- coding: utf-8 -*-
 
 
-def _is_tensor():
-    raise NotImplementedError()
+def _is_tensor(x):
```

Comment:
Things like nested lists could also qualify under this criterion. Could we learn from how other NLA libraries are doing this? I think `np.array` or `np.asarray` may perform this kind of check.